### PR TITLE
Added Table Styling Based on Figma Design for Node.js Website

### LINKED ISSUE #6032: Create `Table` styling
Create table `<table/>` styling using the new [figma design (pagelink)](https://www.figma.com/file/pu1vZPqNIM7BePd6W8APA5/Node.js?type=design&node-id=696-15704&mode=dev). 

This isn't a new React component, just a globally styled HTML element we want to illustrate. The primary use case for this styling is table tags within rendered markdown/MDX files.

> [Read more about the Node.js Website redesign](https://github.com/nodejs/nodejs.org/discussions/5131)

---

### Before You Start...

- Use the [Contributor's Guide](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) to learn how to [setup your environment](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#getting-started), run [available commands](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#cli-commands), and construct [proper commits](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#commit-guidelines).
- Review the [technologies used](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#technologies-used-in-the-website) when building the site, and [why they were chosen](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#remarks-on-technologies-used).

---

### Details

image
image

❓ [Not sure where to start breaking down the figma? Read this short guide](https://github.com/nodejs/nodejs.org/discussions/5131#discussioncomment-6848378)

- Add the new styles to https://github.com/nodejs/nodejs.org/blob/main/styles/new/base.css
- Add these table styles to a new [storybook story file `components/__design__/table.stories.tsx`](https://github.com/nodejs/nodejs.org/blob/main/components/__design__/table.stories.tsx) which will excercise each of the component's states.
- Use the new tailwind color and typography system to assign properties. For example, the Dark mode border `color` uses Neutral/800 per the figma.

There are 2 states to capture within styles and stories:

- Light
- Dark

### Open Questions

Could we make this an `Table` component and then add it to an MDX provider? That might isolate markdown-driven use cases more, but at the cost of increased complexity. For now, let's show what an `<table/>` tag renders like.

The figma design does not differentiate header rows from data rows. Yes, that might look odd. Let's commit to this design for now and adjust as needed.

### PR DESCRIPTION
**Pull request by shaikahmadnawaz**
## Description
This pull request addresses issue #6032, which involves adding table styling to the Node.js website based on the new Figma design. The primary goal is to globally style HTML `<table>` elements to match the design provided in the Figma link.

## Validation

image
image

## Related Issues

Fixes #6032 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by bmuenzenmeyer — CHANGES_REQUESTED** [review]
this is missing border-radius around the table - as a first pass

**Comment by shaikahmadnawaz** [comment]
> this is missing border-radius around the table - as a first pass

I am trying

**Comment by shaikahmadnawaz** [comment]
> > this is missing border-radius around the table - as a first pass
> 
> I am trying for 30 minutes to implement that.

**Comment by shaikahmadnawaz** [comment]
> this is missing border-radius around the table - as a first pass

I added border-radius to the table, how to hide the border overflow of the blocks?

Screenshot 2023-10-20 191059

**Review by shaikahmadnawaz — COMMENTED** [review]
added the border-radius @bmuenzenmeyer 

image
image

File: styles/new/base.css [hunk 1]
```diff
@@ -41,3 +41,50 @@ ol {
     text-neutral-900
     dark:text-white;
 }
+
+table {
+  @apply w-full 
+  border-separate
```
Comment (araujogui) [inline styles/new/base.css]: Please indent styles as shown here: [Example of a CSS Module](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#example-of-a-css-module)

```suggestion
    border-separate 
```

File: styles/new/base.css [hunk 2]
```diff
@@ -41,3 +41,50 @@ ol {
...
+  @apply w-full 
+  border-separate 
+  border-spacing-0 
+  rounded
+  border
+  border-neutral-200 
+  text-left 
+  dark:border-neutral-800;
+
+  th {
```
Comment (araujogui) [inline styles/new/base.css]: `th` and `td` share a lot of properties. Can u create a common selector `th, td`? DRY

File: styles/new/base.css [hunk 3]
```diff
@@ -41,3 +41,50 @@ ol {
...
+
+table {
+  @apply w-full 
+  border-separate 
+  border-spacing-0 
+  rounded
+  border
+  border-neutral-200 
+  text-left 
+  dark:border-neutral-800;
```
Comment (araujogui) [inline styles/new/base.css]: Figma prototype use 14px font size.

Comment (araujogui) [inline styles/new/base.css:53]:
```suggestion
  dark:border-neutral-800;
```

Comment (araujogui) [inline styles/new/base.css]: Figma prototype use 14px font size.
```suggestion
  dark:border-neutral-800
  text-sm;
```

**Review by araujogui — CHANGES_REQUESTED** [review]

File: components/__design__/table.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
```
Comment (araujogui) [inline components/__design__/table.stories.tsx:1]: Please follow this [template](https://github.com/nodejs/nodejs.org/blob/main/COLLABORATOR_GUIDE.md#general-guidelines-for-storybooks)

**Review by araujogui — DISMISSED** [review]

File: components/__design__/table.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+export const Table: StoryObj = {
+  render: () => (
```
Comment (ovflowd) [inline components/__design__/table.stories.tsx:4]: Can you add a 2nd example of a headless table? A table without header?

Comment (shaikahmadnawaz) [reply components/__design__/table.stories.tsx:4]: Shall I add it in the same file table.stories.tsx or in another new file?

Comment (shaikahmadnawaz) [reply components/__design__/table.stories.tsx:4]: tablewithoutheader.stories.tsx @ovflowd

Comment (araujogui) [reply components/__design__/table.stories.tsx:4]: Same file, another Story `HeadlessTable` for example.

Comment (shaikahmadnawaz) [reply components/__design__/table.stories.tsx:4]: image
image

File: styles/new/base.css [hunk 6]
```diff
@@ -41,3 +41,43 @@ ol {
...
+      text-neutral-900 
+      dark:border-neutral-800 
+      dark:text-white;
+  }
+
+  th {
+    @apply font-semibold;
+  }
+
+  tr:last-child {
```
Comment (ovflowd) [inline styles/new/base.css]: This can be simplified to `tr:last-child td`

**Review by ovflowd — APPROVED** [review]
After you fix the latest comment, we're good to merge :)

**Comment by shaikahmadnawaz** [comment]
> After you fix the latest comment, we're good to merge :)

I fixed it.

**Review by bmuenzenmeyer — APPROVED** [review]

**Pull request merged**